test/utils.ts

```diff
@@ -164,3 +166,45 @@ export const SYNTAX_CASES = [
 export const testCompiled = process.env.TEST_COMPILED === '1'
 
 export const srcDir = testCompiled ? 'lib' : 'src'
+
+type RuleTesterOptions = {
+  parser?: string
+} & Omit<TSESLint.RuleTesterConfig, 'parser'>
+
+export class RuleTester extends TSESLint.RuleTester {
+  constructor(option: RuleTesterOptions = {}) {
+    /**
+     * The parser option is required in TSESLint.RuleTester, but not in ESLint.RuleTester.
+     * Since TSESLint.RuleTester is just an alias of ESLint.RuleTester, we can safely ignore the type error.
+     */
+    // @ts-expect-error -- parser is not required
+    super(option)
+  }
```

Comment:
By extending our own `RuleTester` class, we don't have to patch the type of the argument.